release.nix

```diff
@@ -1,34 +1,53 @@
 let
   pkgs = import ./nixpkgs.nix;
 
-  dhall-kubernetes-smoketests =
-    pkgs.runCommand
-      "all-generated-files-compile"
-      { nativeBuildInputs = [
-          pkgs.python3
-          pkgs.bash
-          pkgs.dhall
-          pkgs.glibcLocales
-        ];
-      }
+  nativeBuildInputs = [
+    pkgs.git
+    pkgs.python3
+    pkgs.bash
+    pkgs.dhall
+    pkgs.dhall-json
+    pkgs.dhall-text
+    pkgs.glibcLocales
+  ];
+
+  runCommand = name: script:
+    pkgs.runCommand name { inherit nativeBuildInputs; } script;
+
+  generatedFilesCompile =
+    runCommand
+      "generated-files-compile"
       ''
         cd ${./.}
-        LC_ALL=en_US.UTF-8 python3 ${./check-source.py}
+        LC_ALL=en_US.UTF-8 ./scripts/check-source.py
 
         touch $out
       '';
 
-  # Derivation that trivially depends on the current directory so that Hydra's
-  # pull request builder always posts a GitHub status on each revision
-  pwd = pkgs.runCommand "pwd" { here = ./.; } "touch $out";
+  buildReadme =
+    runCommand
+      "build-readme"
+      ''
+        cd ${./.}
+        LC_ALL=en_US.UTF-8 ./scripts/build-readme.py $out
+
+        diff --unified --color=always README.md $out
```

Comment:
This will not fail the build if the files are different right? If not, should we fail if they don't match?

Reply:
Yes it will. `diff` exits with status code 1 if the files are different.